performance_manager/tests/test_postgres.py

```diff
@@ -30,21 +76,332 @@ def set_environment_variables() -> Iterator[None]:
+            continue
+
+        log = message_to_dict(record.message)
+
+        # check for keys that must be present
+        must_keys = ["status", "uuid", "parent", "process_id", "process_name"]
+        for key in must_keys:
+            if key not in log:
+                pytest.fail(f"Log missing {key} key - {record.message}")
+
+        # process the record, ensuring proper ordering
+        if log["status"] == "complete":
+            # process should be at the end of the stack
+            if process_stack[-1]["uuid"] != log["uuid"]:
+                pytest.fail(
+                    f"Improper Ordering of Log Statements {caplog.text}"
+                )
+            if "duration" not in log:
+                pytest.fail(f"Log missing duration key - {record.message}")
+            process_stack.pop()
+        elif log["status"] == "started":
+            process_stack.append(log)
+
+    if len(process_stack) != 0:
+        pytest.fail(f"Not all processes logged completion {caplog.text}")
 
-    paths_to_load = get_unprocessed_files(
-        "RT_VEHICLE_POSITIONS", db_manager, file_limit=6
+
+def test_static_tables(
+    db_manager: DatabaseManager, caplog: pytest.LogCaptureFixture
+) -> None:
+    """
+    test that static schedule files are loaded correctly into our db
+    """
+    caplog.set_level(logging.INFO)
+
+    paths = [file for file in test_files() if "FEED_INFO" in file]
+    db_manager.add_metadata_paths(paths)
+
+    unprocessed_static_schedules = db_manager.select_as_list(
+        sa.select(MetadataLog.path).where(
+            (MetadataLog.processed == sa.false())
+            & (MetadataLog.path.contains("FEED_INFO"))
+        )
+    )
+
+    process_static_tables(db_manager)
+
+    # these are the row counts in the parquet files computed in a jupyter
+    # notebook without using any of our module. our module should be taking
+    # these parquet files and dropping them into our db directly. after
+    # processing the static tables our db tables should have these many record
+    # counts.
+    row_counts = {
+        StaticTrips: 64879,
+        StaticRoutes: 234,
+        StaticStops: 9706,
+        StaticStopTimes: 1676197,
+        StaticCalendar: 76,
+    }
+
+    with db_manager.session.begin() as session:
+        for table, should_count in row_counts.items():
+            actual_count = session.query(table).count()
+            tablename = table.__tablename__  # type: ignore
+            assert (
+                actual_count == should_count
+            ), f"Table {tablename} has incorrect row count"
+
+    unprocessed_static_schedules = db_manager.select_as_list(
+        sa.select(MetadataLog.path).where(
+            (MetadataLog.processed == sa.false())
+            & (MetadataLog.path.contains("FEED_INFO"))
+        )
     )
 
-    assert len(paths_to_load) == 0
+    assert len(unprocessed_static_schedules) == 0
+
+    check_logs(caplog)
+
+
+def test_gtfs_rt_processing(
```

Comment:
test that all of the individual steps are producing dataframes with the right number of columns and that their lengths make sense.